aya/src/sys/netlink.rs

```diff
@@ -330,7 +338,23 @@ impl NetlinkSocket {
                             // this is an ACK
                             continue;
                         }
-                        return Err(io::Error::from_raw_os_error(-err.error));
+                        let attrs = parse_attrs(&message.data)?;
+                        let err_msg = attrs.get(&NLMSGERR_ATTR_MSG).and_then(|msg| {
+                            CStr::from_bytes_with_nul(msg.data)
+                                .ok()
+                                .map(|s| s.to_string_lossy().into_owned())
+                        });
+                        match err_msg {
```

Comment:
it'd be awesome to get both. How important is it that this function returns io::Error?

Reply:
Not very. I was being lazy and at the same time hesitant to create more error types.